metapackages/auto-instrumentations-node/src/register.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as opentelemetry from '@opentelemetry/sdk-node';
+import { diag, DiagConsoleLogger } from '@opentelemetry/api';
+import {
+  getNodeAutoInstrumentations,
+  getResourceDetectorsFromEnv,
+} from './utils';
+
+diag.setLogger(
+  new DiagConsoleLogger(),
+  opentelemetry.core.getEnv().OTEL_LOG_LEVEL
+);
+
+const sdk = new opentelemetry.NodeSDK({
+  instrumentations: [getNodeAutoInstrumentations()],
+  resourceDetectors: getResourceDetectorsFromEnv(),
+});
+
+try {
+  sdk.start();
+  diag.debug('SDK initialized');
+} catch (error) {
+  diag.error('Error initializing SDK', error);
+}
+
+process.on('SIGTERM', () => {
+  sdk
+    .shutdown()
+    .then(() => diag.debug('SDK terminated'))
+    .catch(error => diag.error('Error terminating SDK', error))
+    .finally(() => process.exit(0));
```

Comment:
Should OpenTelemetry call `process.exit(0)`? is it mandatory?
I guess if the user has more signal handlers it might interrupt with those.

Reply:
I've seen this being used in the SDK's initialization in examples and READMEs, and thought it should handle the same in `SIGTERM`. Maybe removing it would prevent interruptions with the user's code? WDYT?

Reply:
I am not deeply familiar with the details, but It seems wrong to me that a third-party package like this one will call `process.exit()` internally. This should usually be in control of the application.
I guess the READMEs and examples show an example **application** that a user can copy into his code and modify to his needs. Having this embedded in a library is forcing the user into some behaviors which might not fit all applications.